Article/Article.js

```diff
@@ -1,26 +1,131 @@
-// Because classes are not hoised you will need to start your code at the bottom of the page.  Look for the comment "START HERE"
+// Because classes are not hoised you will need to 
+//start your code at the bottom of the page. 
+// Look for the comment "START HERE"
 
 class Article {
   constructor(domElement) {
     // assign this.domElement to the passed in domElement
-    this.domElement;
+    this.domElement = domElement;
     // create a reference to the ".expandButton" class. 
-    this.expandButton;
-    // Using your expandButton reference, update the text on your expandButton to say "expand"
-    
-    // Set a click handler on the expandButton reference, calling the expandArticle method.
-    
+    this.expandButton = domElement.querySelector('.expandButton');
+    // Using your expandButton reference, update the text on your 
+    //expandButton to say "expand"
+    this.expandButton.innerText = 'EXPAND';
+
+    // Set a click handler on the expandButton reference, 
+    //calling the expandArticle method.
+    this.expandButton.addEventListener('click', this.expandArticle.bind(this));
+
   }
 
   expandArticle() {
-    // Using our reference to the domElement, toggle a class to expand or hide the article.
-
+    // Using our reference to the domElement, 
+    //toggle a class to expand or hide the article.
+    this.domElement.classList.toggle('article-open');
+    if (this.domElement.classList.contains('article-open')) {
+      this.expandButton.innerText = "CLOSE";
+    }
+    else if (!this.domElement.classList.contains('article-open')) {
```

Comment:
you don't need an if statement here, since you already did the check on line 25, and `else` already represents the contrary